test/gst-va/encode/vp9.py

```diff
@@ -0,0 +1,169 @@
+  def test(self, case, ipmode, qp, quality, refmode, looplvl, loopshp):
+    self.init(spec, case, ipmode, qp, quality, refmode, looplvl, loopshp)
+    self.encode()
+
+class cbr(VP9EncoderTest):
+  def init(self, tspec, case, gop, bitrate, fps, refmode, looplvl, loopshp):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bitrate   = bitrate,
+      case      = case,
+      fps       = fps,
+      frames    = vars(self).get("brframes", self.frames),
+      gop       = gop,
+      looplvl   = looplvl,
+      loopshp   = loopshp,
+      maxrate   = bitrate,
+      minrate   = bitrate,
+      rcmode    = "cbr",
+    )
+ 
+  @slash.parametrize(*gen_vp9_cbr_parameters(spec))
+  def test(self, case, gop, bitrate, fps, refmode, looplvl, loopshp):
+    self.init(spec, case, gop, bitrate, fps, refmode, looplvl, loopshp)
+    self.encode()
+
+class vbr(VP9EncoderTest):
+  def init(self, tspec, case, gop, bitrate, fps, refmode, quality, looplvl, loopshp):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bitrate   = bitrate,
+      case      = case,
+      fps       = fps,
+      frames    = vars(self).get("brframes", self.frames),
+      gop       = gop,
+      looplvl   = looplvl,
+      loopshp   = loopshp,
+      maxrate   = bitrate,
+      minrate   = bitrate,
+      quality   = quality,
+      rcmode    = "vbr",
+    )
+  
+  @slash.parametrize(*gen_vp9_vbr_parameters(spec))
+  def test(self, case, gop, bitrate, fps, refmode, quality, looplvl, loopshp):
+    self.init(spec, case, gop, bitrate, fps, refmode, quality, looplvl, loopshp)
+    self.encode()
+
+class cqp_lp(VP9EncoderLPTest):
+  def init(self, tspec, case, ipmode, qp, quality, refmode, looplvl, loopshp):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      case      = case,
+      gop       = 30 if ipmode != 0 else 1,
+      looplvl   = looplvl,
+      loopshp   = loopshp,
+      quality   = quality,
+      rcmode    = "cqp",
+    )
+
+  @parametrize_with_unused(*gen_vp9_cqp_lp_parameters(spec), ['slices'])
+  def test(self, case, ipmode, qp, quality, slices, refmode, looplvl, loopshp):
+    self.init(spec, case, ipmode, qp, quality, refmode, looplvl, loopshp)
+    self.encode()
+
+class cbr_lp(VP9EncoderLPTest):
+  def init(self, tspec, case, gop, bitrate, fps, refmode, looplvl, loopshp):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bitrate   = bitrate,
+      case      = case,
+      fps       = fps,
+      frames    = vars(self).get("brframes", self.frames),
+      gop       = gop,
+      looplvl   = looplvl,
+      loopshp   = loopshp,
+      maxrate   = bitrate,
+      minrate   = bitrate,
+      rcmode    = "cbr",
+    )
+  
+  @parametrize_with_unused(*gen_vp9_cbr_lp_parameters(spec), ['slices'])
+  def test(self, case, gop, bitrate, fps, slices, refmode, looplvl, loopshp):
+    self.init(spec, case, gop, bitrate, fps, refmode, looplvl, loopshp)
+    self.encode()
```

Comment:
`refmode` unused, too.  Also, remove that parameter from the init function.

Reply:
updated